core/injector/application_injector.hpp

```diff
@@ -319,6 +320,28 @@ namespace kagome::injector {
       return initialized.value();
     };
 
+    // polkadot trie db getter
+    auto get_polkadot_trie_db =
+        [initialized =
+             boost::optional<std::shared_ptr<storage::trie::PolkadotTrieDb>>(
+                 boost::none)](const auto &injector) mutable
+        -> std::shared_ptr<storage::trie::PolkadotTrieDb> {
+      if (initialized) {
+        return initialized.value();
+      }
+      auto storage =
+          injector.template create<std::shared_ptr<storage::LevelDB>>();
+      auto backend = std::make_shared<storage::trie::PolkadotTrieDbBackend>(
+          storage,
+          common::Buffer{blockchain::prefix::TRIE_NODE},
+          common::Buffer{0});
```

Comment:
Replace this with injector call `injector.create<PolkadotTrieDbBackend>`. You are going to need to add bind for PolkadotTrieDbBackend for that.
The reason for doing this is that in case we want to instantiate another class depending on PolkadotTrieDbBackend we would also write make_shared and create second instance of PolkadotTrieDbBackend, whereas only single instance is necessary. With bind approach only one instance will be created